src/Services/FHIR/FhirObservationService.php

```diff
@@ -22,25 +23,36 @@ public function __construct()
 
     public function createObservationResource($id = '', $data = '', $encode = true)
     {
+        // Check Profile type and add its Category | Status | Note
         switch ($data['formdir']) {
             case 'vitals':
                 $coding['coding']['system'] = "http://terminology.hl7.org/CodeSystem/observation-category";
                 $coding['coding']['code'] = "vital-signs";
                 $coding['coding']['display'] = "Vital Signs";
                 $coding['text'] = "Vital Signs";
                 $this->FHIRData['category'] = ['category' => new FHIRCodeableConcept($coding)];
-                $this->addMembers($data);
+                $data['text'] ? $this->FHIRData['note']['note'] = new FHIRAnnotation(['text' => $data['text']]) : null;
+                $this->FHIRData['status'] = 'final';
                 break;
             default:
                 #TODO: Return CapabilityStatement or Outcome Resource
                 break;
         }
+
+        // Check Profile or Member and add/set appropriately
+        if ($data['profile'] == 'vitals') {
+            $this->addMembers($data);
+        } else {
+            $this->setValues($data);
+        }
+
+        //  TODO: Yash Please Refactor this to as usual as its hard to read.
```

Comment:
I want our resources built using the R4 classes. I don't want some built using json and other with classes or half and half. I want consistent resources. Please...
I can be a little more lenient with parsing but for build. Classes!

Reply:
Ya, thats what I said to myself I will refactor all of it to use our R4 classes :)

Reply:
Done 👍  If u wish u can review this [commit](https://github.com/openemr/openemr/pull/3368/commits/354863e7f08006a8e2df60f583b31320b981ef66) :)